### PR TITLE
Automatically avoid logging if UI thread is currently within `update()` invocation

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/IWithLogger.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import com.virtuslab.logger.IEnhancedLambdaLogger;
+
+public interface IWithLogger {
+  IEnhancedLambdaLogger log();
+
+  /**
+   * @return whether the execution on the current thread is in a context where logging should be avoided
+   *         (because e.g. it would lead to a massive spam in the logs)
+   */
+  boolean isLoggingDiscouraged();
```
Comment (PawelLipski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/IWithLogger.java:12]: Negative name... didn't have any good idea for a positive (non-negated one)... pls help :pray:

Comment (mkondratek) [reply frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/IWithLogger.java:12]: `isLoggingPromoted`/`isLoggingEncouraged`?

or (in the reference to https://tools.ietf.org/html/rfc2119 :trollface:)
`shouldLog`

**Review by mkondratek — COMMENTED** [review]
image
:me-gusta:

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseProjectDependentAction.java [hunk 2]
```diff
@@ -8,18 +8,30 @@
...
 import com.virtuslab.gitmachete.frontend.ui.providerservice.GraphTableProvider;
 import com.virtuslab.gitmachete.frontend.ui.providerservice.SelectedGitRepositoryProvider;
+import com.virtuslab.logger.IEnhancedLambdaLogger;
 
 public abstract class BaseProjectDependentAction extends DumbAwareAction implements IWithLogger {
+  @UIEffect
+  private boolean isUpdateInProgressOnUIThread;
+
+  @Override
+  public boolean isLoggingDiscouraged() {
```
Comment (micpiotrowski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseProjectDependentAction.java:22]: maybe just `isLoggingAllowed` and reverse the logic?

**Review by micpiotrowski — APPROVED** [review]

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseProjectDependentAction.java (hunk 2 above)
Comment (PawelLipski) [reply frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseProjectDependentAction.java]: Finally reversed the logic and changed to `isLoggingAcceptable`

**Review by mkondratek — APPROVED** [review]
:+1:

**Pull request merged**